module-4/backend/week_four.md

```diff
@@ -5,15 +5,44 @@ Fork this respository. Answer the questions to the best of your ability. Try to
 Note: When you're done, submit a PR. 
 
 1. What's your favorite project management tool?
+
+Waffle for software dev
+
 2. Why do we create staging environments?
+
+Test how it will look in production while not actually in production or customer facing.
+
 3. What are the characteristics of a good README (in your opinion)?
+
+Badges based on tests, coverage, versions. Good Intro, Installation instructions, Common errors, demos
+
 4. What's one main improvement you're going to make to your code regarding accessibility issues?
+
+Use site without mouse is possible through tabs
+
 5. What are some basic security concerns to be aware of when building applications?
+
+clickjacking, injection, scripting, man in the middle
+
 6. Why is continuous integration helpful/important?
+
+Run test suite automatically after every change makes sure you don't merge broken code or need to address before merging
+
 7. What are some continuous integration tools?
 
+Semaphore, Travis, Jenkins
+
+
 #### Review  
 
 8. What are some characteristics of "good" git workflow?
+
+merge dev or master into feature branch locally, merge into dev or master on github to deal w/ merge conflicts more easily. Another person reviews PRs and merges if in a group. Commit often. 
+
 9. What are the four fundamental concepts of object oriented programming?
+
+Inheritance, Abstraction, Polymorphism, Encapsulation
+
 10. What's a module in Ruby and what's the difference between a class and a module?
+
+A module can be used for similar classes that share a lot of code or to code the differences between similar objects. It assists in the limitation in Ruby that it has only single inheritance but you can include or extend modules in classes.
```

Comment:
yup! also - a module cannot be instantiated, but classes can. modules don't encapsulate state, only behavior.